pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2113,33 +2114,44 @@ private void handlePoliciesUpdates(NamespaceName namespace) {
     }
 
     private void handleDynamicConfigurationUpdates() {
-        pulsar().getPulsarResources().getDynamicConfigResources().getDynamicConfigurationAsync()
-                .thenAccept(optMap -> {
-                    if (!optMap.isPresent()) {
-                        return;
-                    }
-                    Map<String, String> data = optMap.get();
-                    data.forEach((configKey, value) -> {
-                        Field configField = dynamicConfigurationMap.get(configKey).field;
-                        Object newValue = FieldParser.value(data.get(configKey), configField);
-                        if (configField != null) {
-                            Consumer listener = configRegisteredListeners.get(configKey);
-                            try {
-                                Object existingValue = configField.get(pulsar.getConfiguration());
-                                configField.set(pulsar.getConfiguration(), newValue);
-                                log.info("Successfully updated configuration {}/{}", configKey,
-                                        data.get(configKey));
-                                if (listener != null && !existingValue.equals(newValue)) {
-                                    listener.accept(newValue);
+        DynamicConfigurationResources dynamicConfigResources = null;
+        try {
+            dynamicConfigResources = pulsar()
+                    .getPulsarResources()
+                    .getDynamicConfigResources();
```

Comment:
We are changing the async call to sync. Which thread is this method called? Is it possible to cause some dead lock to blocking request response thread?

Reply:
1. This is just to get the object, it will not block the thread.
2. The try catch here is just to catch the situation where getPulsarResources or getDynamicConfigResources returns null, because some unit tests do not initialize pulsarResources or dynamicConfigResources.
3. Try cach is also performed in the original call of updateDynamicServiceConfiguration:
<img width="1010" alt="image" src="https://user-images.githubusercontent.com/19296967/188568999-c3e6aa55-c94d-4e1e-8514-4955b0651d44.png">

Reply:
handleDynamicConfigurationUpdates is executed via zk callbacks:
<img width="1202" alt="image" src="https://user-images.githubusercontent.com/19296967/188569203-83ab4144-c50d-41cd-8eb2-4c6c6d4c9b6c.png">

Reply:
I see.